app/src/main/java/org/fossasia/openevent/app/common/BaseDetailPresenter.java

```diff
@@ -0,0 +1,22 @@
+package org.fossasia.openevent.app.common;
+
+import android.support.annotation.CallSuper;
+
+import org.fossasia.openevent.app.common.contract.presenter.IDetailPresenter;
+
+public  abstract class BaseDetailPresenter<K, V> extends BasePresenter<V> implements IDetailPresenter<K, V> {
+
+    private K id;
```

Comment:
naming key `id` will make it's usage unclear. just pass multi parameters in `attach` and let the implementation save the key values accordingly. It shall reduce two base presenters into one.